### PR TITLE
implement nop and lru row caches—use lrurowcache

### PR DESCRIPTION
**Pull request by jaffee**
This isn't meant for general consumption (we'll want something more
configurable), but @dmibor might be able to use this to work around some OOM
issues.

Would love to get Todd's thoughts around implications here.

helps with #1893 

## Pull request checklist

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [ ] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

**Comment by dmibor** [comment]
prob won't help much though - I was keeping an eye on this as you advised before, but as you can see from https://github.com/pilosa/pilosa/issues/1893 profiles, this cache is like 500MBs, negligible in the grand scheme of things....

**Comment by jaffee** [comment]
uh... hmm. I must have read something wrong, I thought this was a significant percentage. sorry about that. I'll take another crack tomorrow. we have a few other ideas

File: fragment.go [hunk 1]
```diff
@@ -242,7 +242,7 @@ func (f *fragment) openStorage() error {
 
 	// Attach the file to the bitmap to act as a write-ahead log.
 	f.storage.OpWriter = f.file
-	f.rowCache = &simpleCache{make(map[uint64]*Row)}
+	f.rowCache = newLRUBitmapCache(10)
```
Comment (dmibor) [inline fragment.go]: can maxEntries go to configs?

**Comment by dmibor** [comment]
@jaffee can this be merged? quite a useful thing in any case.

**Comment by jaffee** [comment]
We'll have to decide how (and whether) to expose the configuration for this. Probably needs to be per-field in the general case, but we'll need to think about how we can support different cache types and changing the cache size or type. 

Because we've never used a cache with eviction here, I'd also like to do some more testing/verification to ensure it won't break anything subtle. 

Could you give us a sense of how important this feature is to you so we can prioritize?

**Comment by dmibor** [comment]
medium priority - it hurts a lot only in one case, when people are running exploratory queries on large chunks of dataset - it effectively copies everything in heap and dies. Don't do that too often, but would be nice to eliminate one source of instability...

**Pull request closed without merging**